packages/iota/src/validation/transactionEssence.ts

```diff
@@ -13,10 +14,22 @@ import { IValidationResult, mergeValidationResults } from "./result";
  * @returns The validation result.
  */
 export function validateTransactionEssence(transactionEssence: ITransactionEssence, protocolInfo: INodeInfoProtocol
-    ): IValidationResult {
+): IValidationResult {
     const validateInputsResult = validateInputs(transactionEssence.inputs);
     const validateOutputsResult = validateOutputs(transactionEssence.outputs, protocolInfo);
+    let validatePayloadResult: IValidationResult = { isValid: true };
+    let validateTypeResult: IValidationResult = { isValid: true };
 
-    return mergeValidationResults(validateInputsResult, validateOutputsResult);
+    if (transactionEssence.payload) {
```

Comment:
Hint overall: Ordering of validations should be as the order of the fields in the actual object.
The `.type` validation also always goes first, cause if the type if wrong, we don't even care about the rest.